source/expectations/matchers/ut_be_within.tpb

```diff
@@ -0,0 +1,113 @@
+create or replace type body ut_be_within as
+  /*
+  utPLSQL - Version 3
+  Copyright 2016 - 2019 utPLSQL Project
+
+  Licensed under the Apache License, Version 2.0 (the "License"):
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+  */
+
+ member procedure init(self in out nocopy ut_be_within, a_dist ut_data_value, a_is_pct number , a_self_type varchar2 := null) is
+  begin
+    self.dist  := a_dist;
+    self.is_pct := nvl(a_is_pct,0);
+    self.self_type := nvl( a_self_type, $$plsql_unit );
+  end;
+
+  constructor function ut_be_within(self in out nocopy ut_be_within, a_dist number, a_is_pct number) return self as result is
+  begin
+    init(ut_data_value_number(a_dist),a_is_pct);
+    return;
+  end;
+  
+  constructor function ut_be_within(self in out nocopy ut_be_within, a_dist dsinterval_unconstrained, a_is_pct number) return self as result is
+  begin
+    init(ut_data_value_dsinterval(a_dist),a_is_pct); 
+    return;
+  end;
+  
+  constructor function ut_be_within(self in out nocopy ut_be_within, a_dist yminterval_unconstrained, a_is_pct number) return self as result is
+  begin
+    init(ut_data_value_yminterval(a_dist),a_is_pct);
+    return;
+  end;
+  
+  member procedure of_(self in ut_be_within, a_expected number) is
+    l_result ut_be_within := self;
+  begin 
+    l_result.expected := ut_data_value_number(a_expected);
+    l_result.expectation.to_(l_result );        
+  end;
+  
+  member procedure of_(self in ut_be_within, a_expected date) is
+    l_result ut_be_within := self;
+  begin 
+    l_result.expected := ut_data_value_date(a_expected);
+    l_result.expectation.to_(l_result );    
+  end;
+  
+  overriding member function run_matcher(self in out nocopy ut_be_within, a_actual ut_data_value) return boolean is
+    l_result boolean;
+  begin
+    if self.expected.data_type = a_actual.data_type then
```

Comment:
This code is pretty hard to read. Lot of treat calls and nested if statements.

There must be a way to make this code more readable.